### DISCUSSION TITLE
Editors that use tree-sitter ootb?

### DISCUSSION THREAD

**Discussion by MarkuBu** [tree-sitter/tree-sitter#2940] [General]
Hi, is there a list of editors/IDEs that uses tree-sitter natively ootb? Currently I only know Atom and Helix.

**Comment by IWANABETHATGUY** [comment]
nvim used tree-sitter for it's parsing library https://neovim.io/doc/treesitter/

**Comment by MarkuBu** [comment]
As far as I can see it is a Lua plugin. Native Implementation would be nice.

**Comment by Jieiku** [comment]
Sounds like a great thing to add to the documentation someplace.

A list of editors that use treesitter would be awesome!

**Comment by theHamsta** [comment]
> As far as I can see it is a Lua plugin. Native Implementation would be nice.

Neovim uses Lua for the highlevel logic, but it implements the bindings and low level logic itself https://github.com/neovim/neovim/blob/e6f7e038b8bbca487e78ebfc6fe21d6852330623/src/nvim/lua/treesitter.c#L1

**Comment by theHamsta** [comment]
https://github.com/lapce/lapce uses tree-sitter via the Rust bindings

**Comment by feniljain** [comment]
Even [zed.dev](https://zed.dev/) uses tree-sitter

**Comment by sogaiu** [comment]
The upcoming [Emacs 29 series](https://github.com/emacs-mirror/emacs/blob/b73539832d9c4e802925cb8f261a13473da383b3/etc/NEWS#L36-L44) can be built with tree-sitter support.  The question of it being out-of-the-box seems like it will depend on individual distributions.
